src/lib/WalletInfoCollector.ts

```diff
@@ -180,7 +205,7 @@ export default class WalletInfoCollector {
             || NetworkClient.createInstance(Config.networkEndpoint).init();
         WalletInfoCollector._networkInitializationPromise
             .catch(() => delete WalletInfoCollector._networkInitializationPromise);
-        if (walletType === WalletType.BIP39) {
+        if (walletType === WalletType.BIP39 || walletType === WalletType.LEGACY) {
```

Comment:
KeyguardClient only needs to be created if `checkActivity` is true. We can discuss though whether it's actually worth avoiding an unnecessary call.
I'm not entirely sure how expensive this call is, specifically whether it creates a keyguard iframe.